lib/castle/middleware/sensor.rb

```diff
@@ -86,24 +94,41 @@ def find_end_of_head_open(body)
       end
 
       def join_body(response)
-        response.to_enum.reduce('') do |acc, fragment|
-          acc << fragment.to_s
-          acc
-        end
+        response.to_a.map(&:to_s).join
       end
 
       def close_old_response(response)
         response.close if response.respond_to?(:close)
       end
 
-      def app_id_js_tag(env)
-        script_tag("_castle('setAppId', '#{Castle::Middleware.configuration.app_id}');", env)
+      def complete_js_content(env)
+        app_id!(env)
+        identify!(env)
+        secure!(env)
+
+        commands = ["\n", @script_commands.values.join, "\n"].join
```

Comment:
Not sure I like `@script_commands` to be magically set from within the `*!` methods. Those methods should rather return the contents for testability. But don't spend the time now if you feels it's a tricky fix.

Reply:
I agree, I'll remove it. It's not necessary